client/repolist.c

```diff
@@ -115,18 +127,35 @@ TDNFLoadRepoData(
 
         TDNF_SAFE_FREE_MEMORY(pszRepoFilePath);
         pszRepoFilePath = NULL;
-
         /* may have added multiple repos, go to last one */
         while (*ppRepoNext)
             ppRepoNext = &((*ppRepoNext)->pNext);
     }
 
+    dwError = TDNFAllocateMemory(nRepoId + 1, sizeof(char *), (void **)&ppRepoId);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    pRepoParse = pReposAll;
+
+    while (pRepoParse != NULL) {
+        uint32_t hash = hashString(pRepoParse->pszId) % nRepoId;
+        if (ppRepoId[hash] != NULL && strcmp(ppRepoId[hash], pRepoParse->pszId) == 0) {
+            dwError = ERROR_TDNF_DUPLICATE_REPO_ID;
+            BAIL_ON_TDNF_ERROR(dwError);
```

Comment:
We don't need to use hashes. Just loop twice, it's okay, even when it's O(n^2) - there aren't that many repo configs.